CacheMultimap.adoc

```diff
@@ -0,0 +1,79 @@
+= Native cache Mutimap
+
+The goal is to provide Infinispan Native CacheMultimap support. The proposal here is to start with the smallest interface implementation, in order to provide a proper API to the first multimap client, vert-x project.
+However, this API will evolve, and this first implementation must work on Embedded and Server mode.
+This first Multimap won't support duplicate values on the same key. 
+
+[source, java]
+.CacheMultimap.java
+----
+public interface CacheMultimap<K, V> {
+   boolean put(K key, V value); // <1> <2>
+
+   Collection<V> get(K key); // <3>
+
+   boolean remove(K key); //<4>
+
+   boolean remove(K key, V value);
+   
+   void remove(Predicate<V> p); //<5>
```

Comment:
this could return number of removed elements (since remove tells you if that removes anything). `Collection<V> getAndRemove(Predicate<V> p)` would be useful, too, returning the removed entities.
And if I am not mistaken the correct generics for this should be `Predicate<? super V>`.